packages/cli/src/ampx.ts

```diff
@@ -37,16 +42,51 @@ const usageDataEmitter = await new UsageDataEmitterFactory().getInstance(
   dependencies,
 );
 
-attachUnhandledExceptionListeners(usageDataEmitter);
+const telemetryDataEmitter =
+  await new TelemetryDataEmitterFactory().getInstance(dependencies);
+
+attachUnhandledExceptionListeners(usageDataEmitter, telemetryDataEmitter);
 
 verifyCommandName();
 
 const noticesRenderer = new NoticesRenderer(packageManagerController);
-const parser = createMainParser(libraryVersion, noticesRenderer);
-const errorHandler = generateCommandFailureHandler(parser, usageDataEmitter);
+const parser = createMainParser(
+  libraryVersion,
+  noticesRenderer,
+  telemetryDataEmitter,
+);
+
+const initTime = Date.now() - startTime;
+
+// Below is a workaround in order to send data to telemetry when user force closes a prompt (ie with Ctrl+C)
+// without the counter we would send both success and abort data.
+// Trying to do await telemetryDataEmitter.emitAbortion in errorHandler ends up with:
+// Warning: Detected unsettled top-level await
+let telemetryEmitCount = 0;
+// eslint-disable-next-line @typescript-eslint/no-misused-promises
+process.on('beforeExit', async (code) => {
+  if (telemetryEmitCount !== 0) {
+    process.exit(code);
```

Comment:
```suggestion
    return;
```

process is already exiting at this point.